### DISCUSSION TITLE
[F] Table Data Validation on columns

### DISCUSSION THREAD

**Discussion by simplecommerce** [exceljs/exceljs#2461] [Ideas & Proposals]
Great work on a great library!

## 🚀 Feature Proposal

It would be great to be able to apply data validation directly on a column and its rows when creating the table.

## Motivation

In my use case, I have certain table columns that require the user pick from a list of options, and would like to apply that using data validation.

## Example

```
ws.addTable({
  ref: `A1`,
  name: "Data",
  headerRow: true,
  columns: [
    { name: "Name" },
    { 
        name: "Type",
        dataValidation: {
           type: 'list',
           allowBlank: true,
           formulae: ['"One,Two,Three,Four"']
        }
    }
  ],
  rows: [
    ["first row","One"],
    ["second row",""],
    ["third row","Four"]
  ],
  style: {
    theme: 'TableStyleMedium1',
    showRowStripes: true
  }
});
```

Right now I think the only way to achieve this is to use `getCell().dataValidation` or `ws.dataValidations.add`.

Sorry if I am not clear.

**Comment by linghao518** [comment]
`worksheet.dataValidations.add('A2:A9999', { type: 'list', allowBlank: false, formulae: ['"One,Two,Three,Four"'], showErrorMessage: true, errorStyle: 'error', error: 'Please select a valid value from the list' })`

'dataValidations' is work for me

**Comment by simplecommerce** [comment]
@linghao518 I know it works like that, I meant, it would be great to be able to do it from the table definition, look at my example usage idea :)
